### DISCUSSION TITLE
Error building dependencies when installing spaCy

### DISCUSSION THREAD

**Discussion by jlpz-dev** [explosion/spaCy#12067] [Help: Installation] [answered]
[NonLLVM-Output.txt](https://github.com/explosion/spaCy/files/10368077/NonLLVM-Output.txt)
[LLVM-Output.txt](https://github.com/explosion/spaCy/files/10368079/LLVM-Output.txt)

Trying to install spaCy using pip threw an error at me when trying to install the module "thinc". Looking at the logs, it seemed to me that it was failing due to this error:

`[COMMAND] C:\Program Files\LLVM\bin\clang.exe -c C:\Users\jebus\AppData\Local\Temp\pip-install-1ecnkcfl\blis_389e58cdeb254a3780d36af256f47927\blis\_src\config\bulldozer\bli_cntx_init_bulldozer.c -o C:\Users\jebus\AppData\Local\Temp\tmp_f892tp3\bli_cntx_init_bulldozer.o -O3 -std=c99 -D_POSIX_C_SOURCE=200112L -DBLIS_VERSION_STRING="0.7.0" -DBLIS_IS_BUILDING_LIBRARY -Iinclude\windows-x86_64 -I.\frame\3\ -I.\frame\ind\ukernels\ -I.\frame\3\ -I.\frame\1m\ -I.\frame\1f\ -I.\frame\1\ -I.\frame\include -IC:\Users\jebus\AppData\Local\Temp\pip-install-1ecnkcfl\blis_389e58cdeb254a3780d36af256f47927\blis\_src\include\windows-x86_64
        error: [WinError 2] The system cannot find the file specified
        [end of output]`

I realized that it was attempting to use clang by LLVM and I did not have that application installed on my machine. (You can read the logs up to this point using the txt file titled "NonLLVM-Output.txt). I went ahead and installed it even though the documentation for installing spaCy did not indicate that this was needed to install this module. After doing so, I ran the install command again and was presented with a different error (which can be found in the logs inside the "LLVM-output.txt"). At this point, I feel that something is going wrong and can't seem to pinpoint what it could be. Hoping I can get a second eye on this issue.

**Comment by shadeMe** [comment, marked as answer]
LLVM is required to build [BLIS](https://pypi.org/project/blis/), which is a dependency of the Thinc ML library that spaCy relies on. This is usually not a requirement for end-users as we provide pre-built BLIS binaries for most major platforms and OSes, but it seems like you're currently running a 32-bit version of Python.

Could you upgrade to a [64-bit version of Python](https://www.python.org/downloads/windows/), upgrade `pip` to the latest version and try to reinstall spaCy? That ought to automatically download the pre-built binary wheels for Windows.
